mdsuite/experiment/experiment.py

```diff
@@ -401,3 +448,234 @@ def set_mass(self, element: str, mass: float):
+            self.box_array
+        )
+        self.volume = np.prod(self.box_array)
+        # todo look into replacing these properties
+        self.sample_rate = metadata.sample_rate
+        species_list = copy.deepcopy(metadata.species_list)
+        if update_with_pubchempy:
+            update_species_attributes_with_pubchempy(species_list)
+        # store the species information in dict format
+        species_dict = dict()
+        for sp_info in species_list:
+            species_dict[sp_info.name] = {
+                "mass": sp_info.mass,
+                "charge": sp_info.charge,
+                "n_particles": sp_info.n_particles,
+                # legacy: calculators use this list to determine the number of particles
+                "indices": list(range(sp_info.n_particles)),
+                "properties": [prop_info.name for prop_info in sp_info.properties],
+            }
+        self.species = species_dict
+        # assume the same property for each species
+        self.property_groups = next(iter(species_dict.values()))["properties"]
+
+
+def update_species_attributes_with_pubchempy(species_list: List[SpeciesInfo]):
+    """
+    Add information to the species dictionary
+
+    A fundamental part of this package is species specific analysis. Therefore, the
+    Pubchempy package is used to add important species specific information to the
+    class. This will include the charge of the ions which will be used in
+    conductivity calculations.
+
+    """
+    with importlib.resources.open_text(
+        "mdsuite.data", "PubChemElements_all.json"
+    ) as json_file:
+        pse = json.loads(json_file.read())
+
+    # Try to get the species tensor_values from the Periodic System of Elements file
+
+    # set/find masses
+    for sp_info in species_list:
+        for entry in pse:
+            if pse[entry][1] == sp_info.name:
+                sp_info.mass = [float(pse[entry][3])]
+
+                # If gathering the tensor_values from the PSE file was not successful
+    # try to get it from Pubchem via pubchempy
+    for sp_info in species_list:
+        if sp_info.mass is None:
+            try:
+                temp = pcp.get_compounds(sp_info.name, "name")
+                temp[0].to_dict(
+                    properties=[
+                        "atoms",
+                        "bonds",
+                        "exact_mass",
+                        "molecular_weight",
+                        "elements",
+                    ]
+                )
+                sp_info.mass = [temp[0].molecular_weight]
+                log.debug(temp[0].exact_mass)
+            except (ElementMassAssignedZero, IndexError):
+                sp_info.mass = 0.0
+                log.warning(
+                    f"WARNING element {sp_info.name} has been assigned mass=0.0"
+                )
+    return species_list
+
+
+def _species_list_to_architecture_dict(species_list, n_configurations):
```

Comment:
Maybe add a return statement here, is it `{species_name: {property: <property>}}` ?